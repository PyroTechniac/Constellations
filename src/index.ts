import * as Constants from './lib/util/Constants';
import type { Language } from './lib/structures/Language';
import type { Settings } from './lib/settings/structures/Settings';
import type { GatewayDriver } from './lib/settings/gateway/GatewayDriver';
import type { KeyedObject } from './lib/util/SharedTypes';
import type { SettingsUpdateContext } from './lib/settings/structures/SettingsFolder';

export * from './lib/Client';

// #region extensions

export * from './lib/extensions/ConstellationUser';

// #endregion extensions

// #region settings

export * from './lib/settings/gateway/Gateway';
export * from './lib/settings/gateway/GatewayDriver';
export * from './lib/settings/gateway/GatewayStorage';
export * from './lib/settings/schema/Schema';
export * from './lib/settings/schema/SchemaEntry';
export * from './lib/settings/schema/SchemaFolder';
export * from './lib/settings/structures/Settings';
export * from './lib/settings/structures/SettingsFolder';

// #endregion settings

// #region structures

export * from './lib/structures/base/AliasPiece';
export * from './lib/structures/base/AliasStore';
export * from './lib/structures/base/Piece';
export * from './lib/structures/base/Store';
export * from './lib/structures/Language';
export * from './lib/structures/LanguageStore';
export * from './lib/structures/Provider';
export * from './lib/structures/ProviderStore';
export * from './lib/structures/Serializer';
export * from './lib/structures/SerializerStore';
export * from './lib/structures/SQLProvider';

// #endregion structures

// #region utils

export { Constants };
export * from './lib/util/Colors';
export * from './lib/util/ConstellationConsole';
export * from './lib/util/Cron';
export * from './lib/util/Duration';
export * from './lib/util/QueryBuilder';
export * from './lib/util/RateLimit';
export * from './lib/util/RateLimitManager';
export * from './lib/util/RequestHandler';
export * from './lib/util/SharedTypes';
export * from './lib/util/Stopwatch';
export * from './lib/util/Timestamp';
export * from './lib/util/Type';
export * from './lib/util/Utils';

// #endregion utils

// #region augments

/** @internal */
declare module 'discord.js' {
	interface Client {
		gateways: GatewayDriver;
	}

	interface Guild {
		readonly language: Language;
	}

	interface User {
		settings: Settings;
	}

	interface ClientEvents {
		settingsSync: [Settings];
		settingsDelete: [Settings];
		settingsCreate: [Settings, KeyedObject, SettingsUpdateContext];
		settingsUpdate: [Settings, KeyedObject, SettingsUpdateContext];
	}
}

// #endregion augments