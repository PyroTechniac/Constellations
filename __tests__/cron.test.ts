import { Cron } from '../src';

const minutes = new Array(60).fill(0).map((_, i): number => i);
const hours = new Array(24).fill(0).map((_, i): number => i);
const days = new Array(31).fill(0).map((_, i): number => i + 1);
const months = new Array(12).fill(0).map((_, i): number => i + 1);
const dows = new Array(7).fill(0).map((_, i): number => i);

function odds(numbers: number[]): number[] {
	return numbers.filter((num): boolean => Boolean(num % 2));
}

function evens(numbers: number[]): number[] {
	return numbers.filter((num): boolean => !(num % 2));
}

test('pre-defined-@hourly', (): void => {
	const specimine = new Cron('@hourly');

	expect(specimine.normalized).toBe('0 * * * *');
	expect(specimine.minutes).toStrictEqual([0]);
	expect(specimine.hours).toStrictEqual(hours);
	expect(specimine.days).toStrictEqual(days);
	expect(specimine.months).toStrictEqual(months);
	expect(specimine.dows).toStrictEqual(dows);
});

test('pre-defined-@daily', (): void => {
	const specimine = new Cron('@daily');

	expect(specimine.normalized).toBe('0 0 * * *');
	expect(specimine.minutes).toStrictEqual([0]);
	expect(specimine.hours).toStrictEqual([0]);
	expect(specimine.days).toStrictEqual(days);
	expect(specimine.months).toStrictEqual(months);
	expect(specimine.dows).toStrictEqual(dows);
});

test('pre-defined-@weekly', (): void => {
	const specimine = new Cron('@weekly');

	expect(specimine.normalized).toBe('0 0 * * 0');
	expect(specimine.minutes).toStrictEqual([0]);
	expect(specimine.hours).toStrictEqual([0]);
	expect(specimine.days).toStrictEqual(days);
	expect(specimine.months).toStrictEqual(months);
	expect(specimine.dows).toStrictEqual([0]);
});

test('pre-defined-@monthly', (): void => {
	const specimine = new Cron('@monthly');

	expect(specimine.normalized).toBe('0 0 1 * *');
	expect(specimine.minutes).toStrictEqual([0]);
	expect(specimine.hours).toStrictEqual([0]);
	expect(specimine.days).toStrictEqual([1]);
	expect(specimine.months).toStrictEqual(months);
	expect(specimine.dows).toStrictEqual(dows);
});

test('pre-defined-@yearly', (): void => {
	const specimine = new Cron('@yearly');

	expect(specimine.normalized).toBe('0 0 1 1 *');
	expect(specimine.minutes).toStrictEqual([0]);
	expect(specimine.hours).toStrictEqual([0]);
	expect(specimine.days).toStrictEqual([1]);
	expect(specimine.months).toStrictEqual([1]);
	expect(specimine.dows).toStrictEqual(dows);
});

test('pre-defined-@annually', (): void => {
	const specimine = new Cron('@annually');

	expect(specimine.normalized).toBe('0 0 1 1 *');
	expect(specimine.minutes).toStrictEqual([0]);
	expect(specimine.hours).toStrictEqual([0]);
	expect(specimine.days).toStrictEqual([1]);
	expect(specimine.months).toStrictEqual([1]);
	expect(specimine.dows).toStrictEqual(dows);
});

test('every-minute', (): void => {
	const specimine = new Cron('* * * * *');

	expect(specimine.normalized).toBe('* * * * *');
	expect(specimine.minutes).toStrictEqual(minutes);
	expect(specimine.hours).toStrictEqual(hours);
	expect(specimine.days).toStrictEqual(days);
	expect(specimine.months).toStrictEqual(months);
	expect(specimine.dows).toStrictEqual(dows);
});

test('every-other-minute', (): void => {
	const specimine = new Cron('*/2 * * * *');

	expect(specimine.normalized).toBe('*/2 * * * *');
	expect(specimine.minutes).toStrictEqual(evens(minutes));
	expect(specimine.hours).toStrictEqual(hours);
	expect(specimine.days).toStrictEqual(days);
	expect(specimine.months).toStrictEqual(months);
	expect(specimine.dows).toStrictEqual(dows);
});

test('every-other-hour', (): void => {
	const specimine = new Cron('* */2 * * *');

	expect(specimine.normalized).toBe('* */2 * * *');
	expect(specimine.minutes).toStrictEqual(minutes);
	expect(specimine.hours).toStrictEqual(evens(hours));
	expect(specimine.days).toStrictEqual(days);
	expect(specimine.months).toStrictEqual(months);
	expect(specimine.dows).toStrictEqual(dows);
});

test('every-other-day', (): void => {
	const specimine = new Cron('* * */2 * *');

	expect(specimine.normalized).toBe('* * */2 * *');
	expect(specimine.minutes).toStrictEqual(minutes);
	expect(specimine.hours).toStrictEqual(hours);
	expect(specimine.days).toStrictEqual(odds(days));
	expect(specimine.months).toStrictEqual(months);
	expect(specimine.dows).toStrictEqual(dows);
});

test('every-other-month', (): void => {
	const specimine = new Cron('* * * */2 *');

	expect(specimine.normalized).toBe('* * * */2 *');
	expect(specimine.minutes).toStrictEqual(minutes);
	expect(specimine.hours).toStrictEqual(hours);
	expect(specimine.days).toStrictEqual(days);
	expect(specimine.months).toStrictEqual(odds(months));
	expect(specimine.dows).toStrictEqual(dows);
});

test('every-other-day-of-week', (): void => {
	const specimine = new Cron('* * * * */2');

	expect(specimine.normalized).toBe('* * * * */2');
	expect(specimine.minutes).toStrictEqual(minutes);
	expect(specimine.hours).toStrictEqual(hours);
	expect(specimine.days).toStrictEqual(days);
	expect(specimine.months).toStrictEqual(months);
	expect(specimine.dows).toStrictEqual(evens(dows));
});

test('every-monday-through-friday-at-midnight', (): void => {
	const specimine = new Cron('0 0 * * mon-fri');

	expect(specimine.normalized).toBe('0 0 * * 1-5');
	expect(specimine.minutes).toStrictEqual([0]);
	expect(specimine.hours).toStrictEqual([0]);
	expect(specimine.days).toStrictEqual(days);
	expect(specimine.months).toStrictEqual(months);
	expect(specimine.dows).toStrictEqual([1, 2, 3, 4, 5]);
});

test('every-friday-in-july-at-midnight', (): void => {
	const specimine = new Cron('0 0 * jul fri');

	expect(specimine.normalized).toBe('0 0 * 7 5');
	expect(specimine.minutes).toStrictEqual([0]);
	expect(specimine.hours).toStrictEqual([0]);
	expect(specimine.days).toStrictEqual(days);
	expect(specimine.months).toStrictEqual([7]);
	expect(specimine.dows).toStrictEqual([5]);
});

test('every-friday-in-july-and-august-at-midnight', (): void => {
	const specimine = new Cron('0 0 * jul,aug fri');

	expect(specimine.normalized).toBe('0 0 * 7,8 5');
	expect(specimine.minutes).toStrictEqual([0]);
	expect(specimine.hours).toStrictEqual([0]);
	expect(specimine.days).toStrictEqual(days);
	expect(specimine.months).toStrictEqual([7, 8]);
	expect(specimine.dows).toStrictEqual([5]);
});

test('this-instant', (): void => {
	const now = new Date();
	const [min, hour, day, month, dow] = [now.getUTCMinutes(), now.getUTCHours(), now.getUTCDate(), now.getUTCMonth(), now.getUTCDay()];
	const specimine = new Cron('? ? ? ? ?');

	expect(specimine.normalized).toBe(`${min} ${hour} ${day} ${month} ${dow}`);
	expect(specimine.minutes).toStrictEqual([min]);
	expect(specimine.hours).toStrictEqual([hour]);
	expect(specimine.days).toStrictEqual([day]);
	expect(specimine.months).toStrictEqual([month]);
	expect(specimine.dows).toStrictEqual([dow]);
});

test('bad-cron', (): void => {
	expect((): Cron => new Cron('? ?')).toThrow('Invalid Cron Provided');
});

test('valueOf-is-normalized', (): void => {
	const specimine = new Cron('* * * * mon-fri');

	expect(specimine.valueOf()).toBe('* * * * 1-5');
});

test('toString-is-cron', (): void => {
	const specimine = new Cron('* * * * mon-fri');

	expect(specimine.toString()).toBe('* * * * mon-fri');
});