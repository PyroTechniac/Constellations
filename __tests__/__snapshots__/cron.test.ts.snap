// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`every-friday-in-july-and-august-at-midnight 1`] = `
Cron {
  "cron": "0 0 * jul,aug fri",
  "days": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
    24,
    25,
    26,
    27,
    28,
    29,
    30,
    31,
  ],
  "dows": Array [
    5,
  ],
  "hours": Array [
    0,
  ],
  "minutes": Array [
    0,
  ],
  "months": Array [
    7,
    8,
  ],
  "normalized": "0 0 * 7,8 5",
}
`;

exports[`every-friday-in-july-at-midnight 1`] = `
Cron {
  "cron": "0 0 * jul fri",
  "days": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
    24,
    25,
    26,
    27,
    28,
    29,
    30,
    31,
  ],
  "dows": Array [
    5,
  ],
  "hours": Array [
    0,
  ],
  "minutes": Array [
    0,
  ],
  "months": Array [
    7,
  ],
  "normalized": "0 0 * 7 5",
}
`;

exports[`every-minute 1`] = `
Cron {
  "cron": "* * * * *",
  "days": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
    24,
    25,
    26,
    27,
    28,
    29,
    30,
    31,
  ],
  "dows": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
  ],
  "hours": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
  ],
  "minutes": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
    24,
    25,
    26,
    27,
    28,
    29,
    30,
    31,
    32,
    33,
    34,
    35,
    36,
    37,
    38,
    39,
    40,
    41,
    42,
    43,
    44,
    45,
    46,
    47,
    48,
    49,
    50,
    51,
    52,
    53,
    54,
    55,
    56,
    57,
    58,
    59,
  ],
  "months": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
  ],
  "normalized": "* * * * *",
}
`;

exports[`every-monday-through-friday-at-midnight 1`] = `
Cron {
  "cron": "0 0 * * mon-fri",
  "days": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
    24,
    25,
    26,
    27,
    28,
    29,
    30,
    31,
  ],
  "dows": Array [
    1,
    2,
    3,
    4,
    5,
  ],
  "hours": Array [
    0,
  ],
  "minutes": Array [
    0,
  ],
  "months": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
  ],
  "normalized": "0 0 * * 1-5",
}
`;

exports[`every-other-day 1`] = `
Cron {
  "cron": "* * */2 * *",
  "days": Array [
    1,
    3,
    5,
    7,
    9,
    11,
    13,
    15,
    17,
    19,
    21,
    23,
    25,
    27,
    29,
    31,
  ],
  "dows": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
  ],
  "hours": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
  ],
  "minutes": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
    24,
    25,
    26,
    27,
    28,
    29,
    30,
    31,
    32,
    33,
    34,
    35,
    36,
    37,
    38,
    39,
    40,
    41,
    42,
    43,
    44,
    45,
    46,
    47,
    48,
    49,
    50,
    51,
    52,
    53,
    54,
    55,
    56,
    57,
    58,
    59,
  ],
  "months": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
  ],
  "normalized": "* * */2 * *",
}
`;

exports[`every-other-day-of-week 1`] = `
Cron {
  "cron": "* * * * */2",
  "days": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
    24,
    25,
    26,
    27,
    28,
    29,
    30,
    31,
  ],
  "dows": Array [
    0,
    2,
    4,
    6,
  ],
  "hours": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
  ],
  "minutes": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
    24,
    25,
    26,
    27,
    28,
    29,
    30,
    31,
    32,
    33,
    34,
    35,
    36,
    37,
    38,
    39,
    40,
    41,
    42,
    43,
    44,
    45,
    46,
    47,
    48,
    49,
    50,
    51,
    52,
    53,
    54,
    55,
    56,
    57,
    58,
    59,
  ],
  "months": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
  ],
  "normalized": "* * * * */2",
}
`;

exports[`every-other-hour 1`] = `
Cron {
  "cron": "* */2 * * *",
  "days": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
    24,
    25,
    26,
    27,
    28,
    29,
    30,
    31,
  ],
  "dows": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
  ],
  "hours": Array [
    0,
    2,
    4,
    6,
    8,
    10,
    12,
    14,
    16,
    18,
    20,
    22,
  ],
  "minutes": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
    24,
    25,
    26,
    27,
    28,
    29,
    30,
    31,
    32,
    33,
    34,
    35,
    36,
    37,
    38,
    39,
    40,
    41,
    42,
    43,
    44,
    45,
    46,
    47,
    48,
    49,
    50,
    51,
    52,
    53,
    54,
    55,
    56,
    57,
    58,
    59,
  ],
  "months": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
  ],
  "normalized": "* */2 * * *",
}
`;

exports[`every-other-minute 1`] = `
Cron {
  "cron": "*/2 * * * *",
  "days": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
    24,
    25,
    26,
    27,
    28,
    29,
    30,
    31,
  ],
  "dows": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
  ],
  "hours": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
  ],
  "minutes": Array [
    0,
    2,
    4,
    6,
    8,
    10,
    12,
    14,
    16,
    18,
    20,
    22,
    24,
    26,
    28,
    30,
    32,
    34,
    36,
    38,
    40,
    42,
    44,
    46,
    48,
    50,
    52,
    54,
    56,
    58,
  ],
  "months": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
  ],
  "normalized": "*/2 * * * *",
}
`;

exports[`every-other-month 1`] = `
Cron {
  "cron": "* * * */2 *",
  "days": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
    24,
    25,
    26,
    27,
    28,
    29,
    30,
    31,
  ],
  "dows": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
  ],
  "hours": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
  ],
  "minutes": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
    24,
    25,
    26,
    27,
    28,
    29,
    30,
    31,
    32,
    33,
    34,
    35,
    36,
    37,
    38,
    39,
    40,
    41,
    42,
    43,
    44,
    45,
    46,
    47,
    48,
    49,
    50,
    51,
    52,
    53,
    54,
    55,
    56,
    57,
    58,
    59,
  ],
  "months": Array [
    1,
    3,
    5,
    7,
    9,
    11,
  ],
  "normalized": "* * * */2 *",
}
`;

exports[`pre-defined-@annually 1`] = `
Cron {
  "cron": "@annually",
  "days": Array [
    1,
  ],
  "dows": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
  ],
  "hours": Array [
    0,
  ],
  "minutes": Array [
    0,
  ],
  "months": Array [
    1,
  ],
  "normalized": "0 0 1 1 *",
}
`;

exports[`pre-defined-@daily 1`] = `
Cron {
  "cron": "@daily",
  "days": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
    24,
    25,
    26,
    27,
    28,
    29,
    30,
    31,
  ],
  "dows": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
  ],
  "hours": Array [
    0,
  ],
  "minutes": Array [
    0,
  ],
  "months": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
  ],
  "normalized": "0 0 * * *",
}
`;

exports[`pre-defined-@hourly 1`] = `
Cron {
  "cron": "@hourly",
  "days": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
    24,
    25,
    26,
    27,
    28,
    29,
    30,
    31,
  ],
  "dows": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
  ],
  "hours": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
  ],
  "minutes": Array [
    0,
  ],
  "months": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
  ],
  "normalized": "0 * * * *",
}
`;

exports[`pre-defined-@monthly 1`] = `
Cron {
  "cron": "@monthly",
  "days": Array [
    1,
  ],
  "dows": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
  ],
  "hours": Array [
    0,
  ],
  "minutes": Array [
    0,
  ],
  "months": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
  ],
  "normalized": "0 0 1 * *",
}
`;

exports[`pre-defined-@weekly 1`] = `
Cron {
  "cron": "@weekly",
  "days": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
    21,
    22,
    23,
    24,
    25,
    26,
    27,
    28,
    29,
    30,
    31,
  ],
  "dows": Array [
    0,
  ],
  "hours": Array [
    0,
  ],
  "minutes": Array [
    0,
  ],
  "months": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
  ],
  "normalized": "0 0 * * 0",
}
`;

exports[`pre-defined-@yearly 1`] = `
Cron {
  "cron": "@yearly",
  "days": Array [
    1,
  ],
  "dows": Array [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
  ],
  "hours": Array [
    0,
  ],
  "minutes": Array [
    0,
  ],
  "months": Array [
    1,
  ],
  "normalized": "0 0 1 1 *",
}
`;

exports[`toString-is-cron 1`] = `"* * * * mon-fri"`;

exports[`valueOf-is-normalized 1`] = `"* * * * 1-5"`;
